### DISCUSSION TITLE
group-hover applies to all child elements

### DISCUSSION THREAD

**Discussion by danilokorber** [tailwindlabs/tailwindcss#4547] [Ideas]
I have a problem with group-hover. I have this situation:

```
<div id="div1" class="h-64 p-8 bg-gray-100 group hover:bg-yellow-100">
  <div id="div2" class="h-48 p-8 bg-gray-200 group group-hover:bg-green-200 ">
    <div id="div3" class="h-32 p-8 bg-gray-300 group-hover:bg-blue-300">
    </div>
  </div>
</div>
```

When I hover `div1` I want that only `div2` changes its background. `div3` should only change when the mouse pointer hovers `div2`. In other words, `group-hover` should only apply to its 1st level child elements and not children, grandchildren and so on. Is there any way to achive this with pure tailwind? This can be useful to create menus and submenus.

Here you can see what I mean:  https://stackblitz.com/edit/tailwindcss-ar9cnb?file=index.html

**Comment by MichaelAllenWarner** [comment]
I'm bumping this because I think it's worth considering adding a _direct-parent/child_ mechanism to core (i.e., one that uses the child combinator, `>`).

The problem with the `group-` variants is that they use the descendant combinator (` `), and so you must remember not to nest `group` elements. Within a single component you're unlikely to make that mistake, but when you're building higher-order components from lower-order components, it's an _easy_ mistake to make, and—worse—a difficult mistake to diagnose (e.g., the `group` from your card-collection is affecting the `group-hover` in the button in one of your cards, maybe 3 files removed!). For this reason, if Tailwind shipped with a parent/child mechanism, it would be advisable to use it over the `group-` variants whenever possible.

(To answer the OP's question: if you know what you're doing, it's not terribly difficult to make a custom `children:` variant in the Tailwind config that uses the child combinator, so that you can do something like `hover:children:bg-yellow-100` to target only the direct children of the element when it's hovered.)

**Reply by kvnzrch** [reply]
We need that to be part of the core
